Failing CI: code quality cleanup

CI/CD is rightly failing us in the code quality test because we had code quality issues in the initial repo commits from legacy.

We need to go through these issues and clean them up one-by-one.

Currently `flake8` is bombing us with:
```
./glean/tests/test_cli.py:100:55: E231 missing whitespace after ','
./glean/tests/test_cli.py:127:5: F841 local variable 'results' is assigned to but never used
./glean/api/configs.py:1:1: E266 too many leading '#' for block comment
./glean/api/configs.py:226:1: E266 too many leading '#' for block comment
./glean/api/configs.py:232:11: E713 test for membership should be 'not in'
./glean/api/configs.py:279:1: E266 too many leading '#' for block comment
./glean/api/configs.py:359:33: F821 undefined name 'b'
./glean/api/configs.py:359:56: F821 undefined name 'a'
./glean/api/results.py:223:13: E722 do not use bare 'except'
./glean/api/bundles.py:21:1: F401 'os' imported but unused
./glean/api/bundles.py:21:1: F401 'csv' imported but unused
./glean/api/bundles.py:21:10: E401 multiple imports on one line
./glean/api/bundles.py:68:5: E722 do not use bare 'except'
./glean/api/bundles.py:129:9: E266 too many leading '#' for block comment
./glean/api/bundles.py:207:21: E712 comparison to True should be 'if cond is True:' or 'if cond:'
./glean/api/weights.py:44:9: F841 local variable 'header' is assigned to but never used
./glean/api/weights.py:63:9: F841 local variable 'header' is assigned to but never used
./glean/api/weights.py:151:13: E722 do not use bare 'except'
./glean/api/main.py:139:57: E712 comparison to True should be 'if cond is True:' or 'if cond:'
./glean/api/main.py:145:25: E722 do not use bare 'except'
./glean/api/weights_vcv.py:67:5: E266 too many leading '#' for block comment
./glean/api/weights_vcv.py:68:5: E266 too many leading '#' for block comment
./glean/api/weights_vcv.py:76:5: E266 too many leading '#' for block comment
./glean/api/weights_vcv.py:81:5: E266 too many leading '#' for block comment
```

--- a/glean/api/bundles.py
+++ b/glean/api/bundles.py
@@ -64,7 +64,9 @@ def read(filepath, column="rebased", deltamethod=False):
 
     try:
         rootgrp = Dataset(filepath, "r", format="NETCDF4")
-    except Exception:
+    except Exception as ex:
+        import traceback  # CATBELL
+        print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
         print("Error: Cannot read %s" % filepath)
         exit()
 

--- a/glean/api/bundles.py
+++ b/glean/api/bundles.py
@@ -66,7 +66,10 @@ def read(filepath, column="rebased", deltamethod=False):
         rootgrp = Dataset(filepath, "r", format="NETCDF4")
     except Exception as ex:
         import traceback  # CATBELL
-        print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
+
+        print(
+            "".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__))
+        )  # CATBELL
         print("Error: Cannot read %s" % filepath)
         exit()
 

--- a/glean/api/bundles.py
+++ b/glean/api/bundles.py
@@ -18,7 +18,6 @@ __status__ = "Production"
 __version__ = "$Revision$"
 # $Source$
 
-import os, csv
 import numpy as np
 from netCDF4 import Dataset
 from glean.api import configs
@@ -65,7 +64,7 @@ def read(filepath, column="rebased", deltamethod=False):
 
     try:
         rootgrp = Dataset(filepath, "r", format="NETCDF4")
-    except:
+    except Exception:
         print("Error: Cannot read %s" % filepath)
         exit()
 
@@ -126,7 +125,7 @@ def iterate_regions(filepath, column, config={}):
         data = data2 / 1e5
 
     if deltamethod_vcv is not None and not config.get("deltamethod", False):
-        ## Inferred that these were deltamethod files
+        # Inferred that these were deltamethod files
         config["deltamethod"] = True
 
     if config.get("multiimpact_vcv", None) is not None and deltamethod_vcv is not None:
@@ -204,7 +203,7 @@ def iterate_values(years, values, config={}):
 
     if "yearsets" in config and config["yearsets"]:
         yearsets = config["yearsets"]
-        if yearsets == True:
+        if yearsets:
             yearsets = [(2000, 2019), (2020, 2039), (2040, 2059), (2080, 2099)]
 
         for yearset in yearsets:

--- a/glean/api/configs.py
+++ b/glean/api/configs.py
@@ -357,7 +357,9 @@ def csv_sorted(rowstuffs, config):
     else:
         regioncol = names.index("region")
         key = lambda rowstuff: rowstuff[regioncol]
-        cmp = regionorder.index(b) - regionorder.index(a)
+        cmp = (
+            lambda a, b: regionorder.index(b) - regionorder.index(a)
+        )
 
     if cmp is None:
         return sorted(rowstuffs, key=key)

--- a/glean/api/configs.py
+++ b/glean/api/configs.py
@@ -1,4 +1,5 @@
-## Helper functions for reading the configuration
+"""Helper functions for reading the configuration
+"""
 
 import sys
 import os
@@ -223,13 +224,13 @@ def interpret_filenames(argv, config):
     return columns, basenames, transforms, vectransforms
 
 
-## Plural handling
+# Plural handling
 
 
 def is_allregions(config):
     return not (
         "region" in config or "regions" in config
-    ) and not "region" in config.get("file-organize", [])
+    ) and "region" not in config.get("file-organize", [])
 
 
 def get_regions(config, allregions):
@@ -276,7 +277,7 @@ def get_years(config, years):
     return config.get("years", years)
 
 
-## CSV Creation
+# CSV Creation
 
 
 def csv_organize(rcp, ssp, region, year, config):

--- a/glean/api/configs.py
+++ b/glean/api/configs.py
@@ -357,9 +357,7 @@ def csv_sorted(rowstuffs, config):
     else:
         regioncol = names.index("region")
         key = lambda rowstuff: rowstuff[regioncol]
-        cmp = (
-            lambda a, b: regionorder.index(b) - regionorder.index(a)
-        )
+        cmp = lambda a, b: regionorder.index(b) - regionorder.index(a)
 
     if cmp is None:
         return sorted(rowstuffs, key=key)

--- a/glean/api/main.py
+++ b/glean/api/main.py
@@ -144,7 +144,14 @@ def quantiles(argv, config):
                             weight = model_weights[gcm.lower()]
                         except Exception as ex:
                             import traceback  # CATBELL
-                            print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
+
+                            print(
+                                "".join(
+                                    traceback.format_exception(
+                                        ex.__class__, ex, ex.__traceback__
+                                    )
+                                )
+                            )  # CATBELL
                             print(
                                 "Warning: No weight available for %s, so dropping."
                                 % gcm

--- a/glean/api/main.py
+++ b/glean/api/main.py
@@ -136,13 +136,15 @@ def quantiles(argv, config):
 
                 for batch, gcm, iam in data[filestuff][rowstuff]:
                     value = data[filestuff][rowstuff][(batch, gcm, iam)]
-                    if config.get("deltamethod", False) == True:
+                    if config.get("deltamethod", False):
                         value = results.deltamethod_variance(value, config)
 
                     if do_gcmweights:
                         try:
                             weight = model_weights[gcm.lower()]
-                        except:
+                        except Exception as ex:
+                            import traceback  # CATBELL
+                            print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
                             print(
                                 "Warning: No weight available for %s, so dropping."
                                 % gcm

--- a/glean/api/results.py
+++ b/glean/api/results.py
@@ -220,7 +220,7 @@ def sum_into_data(root, basenames, columns, config, transforms, vectransforms):
                         else:
                             data[filestuff][rowstuff][(batch, gcm, iam)] += value
                         observations += 1
-            except:
+            except Exception:
                 print("Failed to read " + fullpath)
                 traceback.print_exc()
                 if debug:

--- a/glean/api/results.py
+++ b/glean/api/results.py
@@ -20,7 +20,6 @@ __version__ = "$Revision$"
 
 import os
 import glob
-import traceback
 import re
 import numpy as np
 from glean.api import configs, bundles
@@ -220,7 +219,9 @@ def sum_into_data(root, basenames, columns, config, transforms, vectransforms):
                         else:
                             data[filestuff][rowstuff][(batch, gcm, iam)] += value
                         observations += 1
-            except Exception:
+            except Exception as ex:
+                import traceback  # CATBELL
+                print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
                 print("Failed to read " + fullpath)
                 traceback.print_exc()
                 if debug:

--- a/glean/api/results.py
+++ b/glean/api/results.py
@@ -221,7 +221,12 @@ def sum_into_data(root, basenames, columns, config, transforms, vectransforms):
                         observations += 1
             except Exception as ex:
                 import traceback  # CATBELL
-                print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
+
+                print(
+                    "".join(
+                        traceback.format_exception(ex.__class__, ex, ex.__traceback__)
+                    )
+                )  # CATBELL
                 print("Failed to read " + fullpath)
                 traceback.print_exc()
                 if debug:

--- a/glean/api/weights.py
+++ b/glean/api/weights.py
@@ -150,5 +150,10 @@ if __name__ == "__main__":
                 print(gcm, weights[gcm.lower()])
             except Exception as ex:
                 import traceback  # CATBELL
-                print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
+
+                print(
+                    "".join(
+                        traceback.format_exception(ex.__class__, ex, ex.__traceback__)
+                    )
+                )  # CATBELL
                 print("Cannot find weight for %s under %s" % (gcm, rcp))

--- a/glean/api/weights.py
+++ b/glean/api/weights.py
@@ -41,7 +41,7 @@ def get_weights_april2016(rcp):
         "rU",
     ) as tsvfp:
         reader = csv.reader(tsvfp, delimiter="\t")
-        header = next(reader)
+        next(reader)
         for row in reader:
             model = row[1].split("_")[0].strip("*").lower()
             weight = float(row[2])
@@ -60,7 +60,7 @@ def get_weights_march2018(rcp):
         "/shares/gcp/climate/BCSD/SMME/SMME-weights/" + rcp + "_SMME_weights.tsv", "rU"
     ) as tsvfp:
         reader = csv.reader(tsvfp, delimiter="\t")
-        header = next(reader)
+        next(reader)
         for row in reader:
             model = row[1].strip("*").lower()
             if "_" in model:
@@ -148,5 +148,7 @@ if __name__ == "__main__":
         for gcm in os.listdir(os.path.join(batchdir, rcp)):
             try:
                 print(gcm, weights[gcm.lower()])
-            except:
+            except Exception as ex:
+                import traceback  # CATBELL
+                print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
                 print("Cannot find weight for %s under %s" % (gcm, rcp))

--- a/glean/api/weights_vcv.py
+++ b/glean/api/weights_vcv.py
@@ -64,8 +64,8 @@ class WeightedGMCDF(object):
 
 
 if __name__ == "__main__":
-    ## Example between R and python
-    ## R:
+    # # Example between R and python
+    # # R:
     # means <- rnorm(10)
     # sds <- rexp(10)
     # weights <- runif(10)
@@ -73,12 +73,12 @@ if __name__ == "__main__":
     # draws <- sapply(1:100000, function(ii) sample(rnorm(10, means, sds), 1, prob=weights))
     # pp <- runif(10)
     # quantile(draws, pp)
-    ## For the values below:
+    # # For the values below:
     # > quantile(draws, pp)
     #  4.261865%   57.54305%   9.961645%    13.1325%    68.3729%   89.93871%   37.68216%   25.06827%    72.6134%   92.35501%
     # -2.70958468  0.77240194 -2.15403320 -1.90146370  1.17428553  1.95475922 -0.06482985 -0.92293638  1.36865349  2.00405179
 
-    ## Python:
+    # Python:
     means = [
         -1.10402809,
         1.91300947,

--- a/glean/quantiles.py
+++ b/glean/quantiles.py
@@ -26,8 +26,7 @@ if __name__ == "__main__":
     from glean.api.configs import consume_config
 
     warnings.warn(
-        "quantiles.py is deprecated, please use `glean quantiles`",
-        FutureWarning
+        "quantiles.py is deprecated, please use `glean quantiles`", FutureWarning
     )
 
     config, argv = consume_config()

--- a/glean/single.py
+++ b/glean/single.py
@@ -14,10 +14,7 @@ if __name__ == "__main__":
     from glean.api import single
     from glean.api.configs import consume_config
 
-    warnings.warn(
-        "single.py is deprecated, please use `glean single`",
-        FutureWarning
-    )
+    warnings.warn("single.py is deprecated, please use `glean single`", FutureWarning)
 
     config, argv = consume_config()
     single(argv, config)

--- a/glean/tests/test_cli.py
+++ b/glean/tests/test_cli.py
@@ -97,7 +97,7 @@ def test_quantiles_nofile(mocker):
             {"abc": 123},
         ),
         (
-            ["-c", "region=CAN.1.2.28", "basefilename", ],
+            ["-c", "region=CAN.1.2.28", "basefilename"],
             ("basefilename",),
             {"abc": 123, "region": "CAN.1.2.28"},
         ),

--- a/glean/tests/test_cli.py
+++ b/glean/tests/test_cli.py
@@ -97,7 +97,7 @@ def test_quantiles_nofile(mocker):
             {"abc": 123},
         ),
         (
-            ["-c", "region=CAN.1.2.28", "basefilename",],
+            ["-c", "region=CAN.1.2.28", "basefilename", ],
             ("basefilename",),
             {"abc": 123, "region": "CAN.1.2.28"},
         ),
@@ -124,5 +124,5 @@ def test_quantiles_argpass(mocker, tempfl, addargs, expected_argv, expected_conf
 
     runner = CliRunner()
 
-    results = runner.invoke(glean.cli.glean_cli, cli_args)
+    runner.invoke(glean.cli.glean_cli, cli_args)
     glean.api.quantiles.assert_called_once_with(expected_argv, expected_config)

